tests/testthat/test-test-threads.R

```diff
@@ -173,57 +186,49 @@ withr::with_options(list(bbr.bbi_exe_path = read_bbi_path()), {
     )
   })
 
-  test_that("test_threads(.dry_run=T) threads are set correctly [BBR-TSTT-003]", {
-    mod_threads <- lapply(mods, function(mod.x){mod.x$bbi_args$threads}) %>% unlist()
-    expect_equal(mod_threads[1], 2)
-    expect_equal(mod_threads[2], 4)
-  })
 
+  test_that("test_threads(.dry_run=T) correctly changes maxeval/niter: handles different spelling [BBR-TSTT-002]", {
+    # recursive test from adding `nmrec` dependency - lower case and different spellings previously not supported
+    mod_complex_path <- mod_complex$absolute_model_path
+    mod_lines <- get_model_path(mod_complex) %>% readLines()
 
-  test_that("get_est_idx() works correctly: models [BBR-TSTT-004]", {
-    mod_lines <- mod_complex %>% get_model_path() %>% readLines()
-    expect_equal(get_est_idx(mod_lines) %>% unlist(), c(38, 39, 40))
+    # replace NBURN=0 with nburn=10 (0 wont be overwritten by test_threads)
+    nburn_loc <- grep("(?i)NBURN", mod_lines)
+    mod_lines[nburn_loc] <- gsub("NBURN=0", "nburn=10", mod_lines[nburn_loc])
+    new_mod_name <- paste0(mod_complex_path, "_fake")
+    writeLines(mod_lines, ctl_ext(new_mod_name))
 
-    mod_lines <- mod_complex2 %>% get_model_path() %>% readLines()
-    expect_equal(get_est_idx(mod_lines) %>% unlist(), c(111:115))
-    expect_equal(get_est_idx(mod_lines)[[1]], c(111:113))
-    expect_equal(get_est_idx(mod_lines)[[2]], c(114:115))
+    mod_complex_edit <- new_model(
+      file.path(MODEL_DIR_BBI, basename(new_mod_name)),
+      .tags = c("test threads"), # for deletion purposes
+      .bbi_args = list(overwrite = TRUE, threads = 2),
+    )
 
-    mod_lines <- mod_complex3 %>% get_model_path() %>% readLines()
-    expect_equal(get_est_idx(mod_lines)[[1]], 50)
+    mods_fake <- test_threads(mod_complex_edit, .threads = c(2, 4), .cap_iterations = 5, .mode = "local", .dry_run = TRUE)
+    max_evals <- get_est_options(c(list(mod_complex, mod_complex_edit), mods_fake))
 
+    for(i in 1:length(max_evals)){
+      max_evals_i <- unlist(unname(max_evals[[i]]))
+      if(i == 1){ # Original model (for clarity)
+        expect_equal(unname(max_evals_i), c(9999, 10, 0))
+        expect_equal(names(max_evals_i), c("MAXEVAL", "NITER", "NBURN"))
+      }else if (i==2){ # modified model (NBURN=0 --> nburn=10)
+        expect_equal(unname(max_evals_i), c(9999, 10, 10))
+        expect_equal(names(max_evals_i), c("MAXEVAL", "NITER", "nburn"))
+      }else{ # created with test_threads (capped iterations)
+        expect_equal(unname(max_evals_i), c(5, 5, 5))
+        expect_equal(names(max_evals_i), c("MAXEVAL", "NITER", "nburn"))
+      }
+    }
+    delete_models(c(list(mod_complex_edit), mods_fake), .force = T)
```

Comment:
That `list` is unnecessary, right? At a glance, think that can just be `c(mod_complex_edit, mods_fake)`.

Reply:
No `delete_models` expects model objects or lists:
```r
>     delete_models(c(mod_complex_edit, mods_fake), .force = T)
Error in check_model_object_list(.mods, .mod_types = c(NM_MOD_CLASS, NM_SUM_CLASS)) :
Passed list must contain only model objects, but found 5 invalid objects at indices: 1, 2, 3, 4, 5

>     delete_models(c(list(mod_complex_edit), mods_fake), .force = T)
Removed 3 models with the following tags:
- test threads
```

Reply:
Doh, because model object is a list underneath.  Thanks for the correction.